lib/crystalball/map_generator/configuration.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Crystalball
+  class MapGenerator
+    # Configuration of map generator. Is can be accessed as a first argument inside
+    # `Crystalball::MapGenerator.start! { |config| config } block.
+    class Configuration
+      attr_writer :execution_detector, :map_storage
+
+      def execution_detector
+        @execution_detector ||= ExecutionDetector.new(Dir.pwd)
+      end
+
+      def map_storage_path
+        @map_storage_path ||= Pathname('execution_map.yml')
```

Comment:
I wonder if we really need this default value :)

Reply:
I'd guess that most users wouldn't care enough to customize it, so it's always better to have some default for configuration